wd_helper.au3

```diff
@@ -636,6 +636,156 @@ Func _WD_FrameLeave($sSession)
 	Return SetError(__WD_Error($sFuncName, $_WD_ERROR_Success), 0, $sValue)
 EndFunc   ;==>_WD_FrameLeave
 
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_FrameList
+; Description ...: get frame list from current webdriver context/tab
+; Syntax ........: _WD_FrameList($sSession[, $bReturnAsArray = True[, $sFilter = ''[, $bReturnHTML = False]]])
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $bReturnAsArray      - By default on True function will return array, other wise string
+;                  $sFilter             - RegExp pattern to check if desired content exist in frame document HTML source, if used it returns only frames with diesired content
+;                  $bReturnHTML         - On true function will return frame document HTML source
+; Return values .: array or string or set @error to $_WD_ERROR_Exception, $_WD_ERROR_InvalidExpression, $_WD_ERROR_NotFound
+; Author ........: mLipok
+; Modified ......:
+; Remarks .......:
+; Related .......: __WD_FrameList_Internal, _WD_FrameEnter, _WD_GetFrameCount
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_FrameList($sSession, $bReturnAsArray = True, $sFilter = '', $bReturnHTML = False)
+	Local $iErr = $_WD_ERROR_Success
+	Local $a_Result[0][6]
+	Local $sStartBody_ElementID = _WD_ExecuteScript($sSession, "return window.document.body;", Default, Default, $_WD_JSON_Element)
+
+	Local $vResult = __WD_FrameList_Internal($sSession, 'null', '', $sFilter, $bReturnHTML)
+	$iErr = @error
+	#Region - _WD_FrameList - post processing
+	If $iErr = $_WD_ERROR_Success Then ; can occur when $sFilter is used
+		$vResult = StringTrimRight($vResult, 2) ; last @CRLF
+		If $vResult = '' Then
+			$iErr = $_WD_ERROR_NotFound
+		Else
+			_ArrayAdd($a_Result, $vResult)
+
+			Local $sStartLocation
+			For $i = 0 To UBound($a_Result) - 1
+				If $a_Result[$i][4] = $sStartBody_ElementID Then $sStartLocation = $a_Result[$i][0]
+			Next
+
+			For $i = 0 To UBound($a_Result) - 1
+				$a_Result[$i][1] = StringRegExpReplace($a_Result[$i][1], '\A' & $sStartLocation & '\/?', '')
+			Next
+
+			If $bReturnAsArray Then
+				$vResult = $a_Result
+			Else
+				$vResult = _ArrayToString($a_Result)
+			EndIf
+		EndIf
+	EndIf
+
+	; Back to "calling frame"
+	_WD_FrameEnter($sSession, $sStartBody_ElementID)
+	#TODO decide if Error handling here is required or not ?
```

Comment:
@Danp2  what you think abou Error handling here ?

Reply:
Please take a look on: [error handling + headers](https://github.com/Danp2/au3WebDriver/pull/362/commits/74f6fb0c88c2cfd426b9e10b6604731e77b77c03)

> I would think that you need some form of error checking here because _WD_FrameEnter can generate an error.

suplemented

> I'm confused by this implementation. Why would you be calling _WD_FrameEnter with the body element as a target?

fixed